### PR TITLE
Fix all instances of georeference

### PR DESCRIPTION
**Pull request by MishkaMN**
# PR Details
## Description
This PR supports https://github.com/usdot-fhwa-stol/carma-platform/pull/2403
It fixes all instances of georeference processing callbacks as they might run into same issue.
I separated the 2 PRs just to make it digestable

## Related GitHub Issue
https://github.com/usdot-fhwa-stol/carma-platform/issues/2402

## Related Jira Key
https://usdot-carma.atlassian.net/browse/CAR-6059

## Motivation and Context
Seen it simulation as well as live vehicle.
Distributed Testing (VOICES) relies on vehicle's valid BSM, so encountered this issue.

## How Has This Been Tested?
Tested on Black Pacifica and I didn't see any BSM errors for 5+ runs where I was seeing almost every run.

## Types of changes

- [X] Defect fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [X] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING**](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) document.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

File: traffic_incident_parser/src/traffic_incident_parser_worker.cpp [hunk 1]
```diff
@@ -282,14 +282,19 @@ namespace traffic_incident_parser
             RCLCPP_WARN_STREAM(logger_->get_logger(), "Traffic Incident Parser received traffic control message, but it has not loaded the map yet. Returning empty list");
             return {};
         }
+        if (projection_msg_ == "")
+        {
+            RCLCPP_WARN_STREAM(logger_->get_logger(), "Traffic Incident Parser received traffic control message, but georeference has not loaded yet. Returning empty list");
```
Comment (JonSmet) [inline traffic_incident_parser/src/traffic_incident_parser_worker.cpp:287]: This should be "... is composing a Traffic Control Message, but ...", would you mind updating it here and in the above warn statement as well?

**Review by JonSmet — CHANGES_REQUESTED** [review]

File: traffic_incident_parser/src/traffic_incident_parser_worker.cpp (hunk 1 above)
Comment (MishkaMN) [reply traffic_incident_parser/src/traffic_incident_parser_worker.cpp]: Revised

**Comment by MishkaMN** [comment]
Fixed all the failing packages, and currently only the cpplint in motion_computation is failing which is being handled in other PR. @JonSmet

**Comment by MishkaMN** [comment]
```
1: [INFO] [1718392630.629372906] [carma_node]: Destroying
3: /opt/carma/src/carma-platform/motion_computation/src/motion_computation_worker.cpp:152:  Lines should be <= 100 characters long  [whitespace/line_length] [2]
3: /opt/carma/src/carma-platform/motion_computation/src/motion_computation_worker.cpp:157:  Lines should be <= 100 characters long  [whitespace/line_length] [2]
3: Category 'whitespace/line_length' errors found: 2
3: Total errors found: 2
```
This was the error this time which I fixed

**Review by JonSmet — APPROVED** [review]
Approved with unit test coverage below 80% since we are resolving a potential race condition on a code block that is duplicated across our codebase. Unit tests for georeference operations could be addressed as part of future testing improvements.

**Pull request merged**